framework/db/sqlite/Command.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\db\sqlite;
+
+use yii\db\SqlToken;
+use yii\helpers\StringHelper;
+
+/**
+ * Command represents an SQLite's SQL statement to be executed against a database.
+ *
+ * {@inheritdoc}
+ *
+ * @author Sergey Makinen <sergey@makinen.ru>
+ * @since 2.0.14
+ */
+class Command extends \yii\db\Command
+{
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        $sql = $this->getSql();
+        $params = $this->params;
+        $statements = $this->splitStatements($sql, $params);
+        if ($statements === false) {
+            return parent::execute();
+        }
+
+        $result = null;
+        foreach ($statements as $statement) {
+            list($statementSql, $statementParams) = $statement;
+            $this->setSql($statementSql)->bindValues($statementParams);
+            $result = parent::execute();
+        }
+        $this->setSql($sql)->bindValues($params);
+        return $result;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function queryInternal($method, $fetchMode = null)
+    {
+        $sql = $this->getSql();
+        $params = $this->params;
+        $statements = $this->splitStatements($sql, $params);
+        if ($statements === false) {
+            return parent::queryInternal($method, $fetchMode);
+        }
+
+        list($lastStatementSql, $lastStatementParams) = array_pop($statements);
+        foreach ($statements as $statement) {
+            list($statementSql, $statementParams) = $statement;
+            $this->setSql($statementSql)->bindValues($statementParams);
+            parent::execute();
+        }
+        $this->setSql($lastStatementSql)->bindValues($lastStatementParams);
+        $result = parent::queryInternal($method, $fetchMode);
+        $this->setSql($sql)->bindValues($params);
+        return $result;
+    }
+
+    /**
+     * Splits the specified SQL code into individual SQL statements and returns them
+     * or `false` if there's a single statement.
+     * @param string $sql
+     * @param array $params
+     * @return string[]|false
+     */
+    private function splitStatements($sql, $params)
+    {
+        $semicolonIndex = strpos($sql, ';');
```

Comment:
There could be semicolon in string content:

INSERT INTO post (content) VALUES ('Hello; world!');

Reply:
It's okay because it is a "fail fast" condition to avoid parsing at all.